### DISCUSSION TITLE
Indentation python

### DISCUSSION THREAD

**Discussion by DavidHHDS** [LazyVim/LazyVim#4932] [Extra Requests]
hi i am using ruff to format python code with 2 indentation spaces but this only works when formatting code not while typing, when you type it automatically inserts 4, is there a way to set indentation for python to 2 spaces?

**Comment by dpetka2001** [comment]
I believe that comes from Neovim Python ftplugin. You could overwrite the settings in `FileType` autocmd and change it to what you want. `:h expandtab`, `:h shiftwidth` and `:h tabstop` should give you more info.

**Reply by DavidHHDS** [reply]
Thanks, I managed to fix it by adding this line init.lua:
vim.cmd [[autocmd FileType python set shiftwidth=2 tabstop=2 expandtab]]

**Comment by Gao123861** [comment]
Perhaps you can use the backslash(\) to escape special characters in the string, including spaces.
